week4-ts/src/components/SearchSection.tsx

```diff
@@ -0,0 +1,87 @@
+import styled from "styled-components";
+import { useState, useEffect } from "react";
+
+interface SearchSectionProps {
+  onClick: (isCheck: boolean, myTown: string) => void;
+}
+
+export default function SearchSection({ onClick }: SearchSectionProps) {
+  const [isCheck, setIsCheck] = useState<boolean>(false);
+  const [myTown, setMyTown] = useState<string>("");
+
+  useEffect(() => {
+    if (isCheck) setMyTown("");
+  }, [isCheck]);
+
+  return (
+    <StWrapper>
+      <StCheckBoxWrapper>
+        <label htmlFor="checkbox">지역 기반으로 검색</label>
+        <input
+          type="checkbox"
+          id="checkbox"
+          onChange={() => setIsCheck(!isCheck)}
+        />
+      </StCheckBoxWrapper>
+
+      <StTextInputWrapper>
+        <label htmlFor="text">우리 동네는 여기에요</label>
+        <input
+          type="text"
+          placeholder="지역을 입력해주세요."
+          disabled={isCheck}
+          value={myTown}
+          onChange={(e) => setMyTown(e.target.value)}
+        />
+      </StTextInputWrapper>
+
+      <StSearchButton type="button" onClick={() => onClick(isCheck, myTown)}>
+        검색하기
+      </StSearchButton>
+    </StWrapper>
+  );
+}
+
+const StWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding: 10px 0;
+  gap: 10px;
+  font-size: 20px;
+  color: #fff;
+  border-bottom: 2px solid #fff;
+  width: 100%;
+`;
```

Comment:
이 부분 form테그로 wrapping해서 사용하면 좋을 것 같아요!
```suggestion
   return (
    <StForm onSubmit={(e) => onClick(isCheck, myTown, e)}>
      <StCheckBoxWrapper>
        <label htmlFor="checkbox">지역 기반으로 검색</label>
        <input
          type="checkbox"
          id="checkbox"
          onChange={() => setIsCheck(!isCheck)}
        />
      </StCheckBoxWrapper>

      <StTextInputWrapper>
        <label htmlFor="text">우리 동네는 여기에요</label>
        <input
          type="text"
          placeholder="지역을 입력해주세요."
          disabled={isCheck}
          value={myTown}
          onChange={(e) => setMyTown(e.target.value)}
        />
      </StTextInputWrapper>

      <StSearchButton type="submit">
        검색하기
      </StSearchButton>
    </StForm>
  );
}

const StForm = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px 0;
  gap: 10px;
  font-size: 20px;
  color: #fff;
  border-bottom: 2px solid #fff;
  width: 100%;
`;
```

Reply:
근데 form tag를 쓰고 onSubmit이벤트를 발생시키면 디폴트로 브라우저에서 새로고침하는 동작을 넣어버려서 onSubmit이벤트 내부에 들어있는 함수에서 그런 불필요한 디폴트 동작을 막아줘야 하는데 (SPA에서 새로고침은 불필요한 동작이니까여 !) 그러기 위해서 
1. 일단 event인자를 받아주도록 수정하고,

Reply:
아 7주차 과제에서 이런 식으로 짰었는데 이것도 그렇게 바꿔볼게요! 
이 방법이 더 좋아보이네요